### ISSUE TITLE
Journalbeat repeat old message after logs rotation

### ISSUE THREAD

**Issue by jlemksy** [elastic/beats#11758]
After journald rotates logs Journalbeat send the last message from rotated logs on each new event in the journal.

For example:
On start, journalbeat reads all messages which presented in journald and follow journald. After rotation new file system@1.journal created. Now only last event from this file will be sent every time new event appears in journald. After second rotation journalbeat start reading system@2.journal which contain new events. When all events were read Journalbeat again sends the last event from system@2.journal on each new event in journald.

Journalbeat 6.7.0, 6.7.1
Systemd: systemd 237
+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD -IDN2 +IDN -PCRE2 default-hierarchy=hybrid
Operating System: Ubuntu 18.04.1 LTS (Bionic Beaver)
Discuss Forum URL: https://discuss.elastic.co/t/journalbeat-6-7-repeats-old-log-records-after-rotation/174940

journalbeat.yml
```
journalbeat.inputs:
- paths: []
  seek: cursor

output.logstash:
  hosts: ["x.x.x.x:5045"]
  max_retries: -1
 
processors:
  - add_host_metadata: ~
  - add_cloud_metadata: ~
```

Right now I downgraded Journalbeat to 6.6.2 and forced to restart him on every rotation.

**Comment by rlueckl** [comment]
I have a very similar issue where Journalbeat repeats the last line of a journal endlessly, even without rotating the journal. So a forced restart after rotation doesn't really help me.

journalbeat.yml
```
journalbeat.inputs:
- paths: []
  seek: cursor
  include_matches:
    - "systemd.unit=customapp.service"

output.logstash:
  hosts: ["my.logstash.host:1234"]
  ssl.certificate_authorities: ["/etc/ssl/certs/internalCA.pem"]

processors:
  - add_host_metadata: ~
  - add_cloud_metadata: ~
```

In Logstash I see the last line repeated every few seconds even though nothing is happening on the server. "customapp" writes nothing to the journal after starting up. The only thing that changes is _read_timestamp_, otherwise the messages are identical.

Journalbeat 6.7.1, 6.8.2
OS: Debian GNU/Linux 9.9 (stretch)
Systemd: 232-25+deb9u11

**Comment by rlueckl** [comment]
I did some more testing and I *don't* have the issue with the last line repeating indefinitely with Versions 6.6.2 and 6.5.0 - so this seems to be a problem in >=6.7.0

Now, I've forced a rotation of the journal with `journalctl --rotate` and journalbeat doesn't send *anything* anymore. If I restart journalbeat it'll pick up where it was before the rotation and sends everything since.
6.7.1 continues sending logs even after `journalctl --rotate` (with the downside that it repeats the last line...).

So basically both (>=6.7.0 and <6.7.0) are broken, but they are broken differently.

**Comment by timp87** [comment]
Hi! I can't observe this with journalbeat 6.8.8

**Comment by timp87** [comment]
Up

**Comment by jlind23** [comment]
Hello @jlemksy  

As of 7.16 journald is provided as an input to Filebeat: https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-journald.html which supports multi-line. Consequently journalbeat has been deprecated from that release onwards.
So I think we can close this issue.

cc @kvch
